### ISSUE TITLE
Crash due to isFinalizable test left behind for non-contiguous stack allocated object

### ISSUE THREAD

**Issue by hzongaro** [eclipse/openj9#4885]
We're seeing an intermittent crash running the `TreeMapFactory` system test.

**Comment by hzongaro** [comment]
The problem appears to be with an inline `isFinalizable` test that's left behind for an object that was made discontiguous, stack-allocated by Escape Analysis.

Before Escape Analysis we have these trees:

```
n2516n    treetop                                                                             [0x7f80f9501230] bci=[30,4,12] rc=0 vc=1186 vn=471 li=- udi=- nc=1
n2515n      new  jitNewObject[#88  helper Method] [flags 0x400 0x0 ]                          [0x7f80f95011e0] bci=[30,4,12] rc=3 vc=1186 vn=75 li=- udi=- nc=1
n2514n        loadaddr  java/lang/Integer[#704  Static] [flags 0x18307 0x0 ]                  [0x7f80f9501190] bci=[30,4,12] rc=1 vc=1186 vn=74 li=- udi=- nc=0
n2659n    astore  <temp slot 6>[#728  Auto] [flags 0x7 0x0 ]                                  [0x7f80f9503ee0] bci=[30,4,12] rc=0 vc=1186 vn=75 li=102 udi=30 nc=1
n2515n      ==>new
n2638n    astore  <temp slot 3>[#725  Auto] [flags 0x20000007 0x0 ] (privatizedInlinerArg sharedMemory )  [0x7f80f9503850] bci=[30,4,12] rc=0 vc=1186 vn=75 li=103 udi=31 nc=1 flg=0x2000
n2515n      ==>new
...
n2613n    ificmpne --> block_249 BBStart at n2619n ()                                         [0x7f80f9503080] bci=[33,0,39] rc=0 vc=1186 vn=478 li=- udi=- nc=2 flg=0x20
n3407n      iand                                                                              [0x7f80f95b28b0] bci=[33,0,39] rc=1 vc=1186 vn=86 li=- udi=- nc=2
n3406n        l2i                                                                             [0x7f80f95b2860] bci=[33,0,39] rc=1 vc=1186 vn=85 li=- udi=- nc=1
n2609n          lloadi  <isClassAndDepthFlags>[#367  Shadow +24] [flags 0x603 0x0 ]           [0x7f80f9502f40] bci=[33,0,39] rc=1 vc=1186 vn=84 li=- udi=- nc=1
n2608n            aloadi  <vft-symbol>[#370  Shadow] [flags 0x18607 0x0 ] (X>=0 sharedMemory )  [0x7f80f9502ef0] bci=[33,0,39] rc=1 vc=1186 vn=83 li=- udi=- nc=1 flg=0x100
n2604n              aload  <temp slot 3>[#725  Auto] [flags 0x20000007 0x0 ] (X!=0 X>=0 sharedMemory )  [0x7f80f9502db0] bci=[33,0,39] rc=1 vc=1186 vn=75 li=255 udi=196 nc=0 flg=0x104
n3404n        iconst 0x40200000                                                               [0x7f80f95b27c0] bci=[33,0,39] rc=1 vc=1186 vn=82 li=- udi=- nc=0
n3405n      iconst 0                                                                          [0x7f80f95b2810] bci=[33,0,39] rc=1 vc=1186 vn=4 li=- udi=- nc=0
```

After Escape Analysis, we're left with these trees, in part:

```
n4486n    astore  <temp slot 14>[#816  Auto] [flags 0x20000007 0x0 ]                          [0x7f80f96879f0] bci=[30,4,12] rc=0 vc=0 vn=- li=- udi=- nc=1
n4485n      aconst NULL                                                                       [0x7f80f96879a0] bci=[30,4,12] rc=1 vc=0 vn=- li=- udi=- nc=0
...
n2613n    ificmpne --> block_249 BBStart at n2619n ()                                         [0x7f80f9503080] bci=[33,0,39] rc=0 vc=1186 vn=- li=- udi=- nc=2 flg=0x20
n3407n      iand                                                                              [0x7f80f95b28b0] bci=[33,0,39] rc=1 vc=1186 vn=- li=- udi=- nc=2
n3406n        l2i                                                                             [0x7f80f95b2860] bci=[33,0,39] rc=1 vc=1186 vn=- li=- udi=- nc=1
n2609n          lloadi  <isClassAndDepthFlags>[#367  Shadow +24] [flags 0x603 0x0 ]           [0x7f80f9502f40] bci=[33,0,39] rc=1 vc=1186 vn=- li=- udi=- nc=1
n2608n            aload  <temp slot 14>[#816  Auto] [flags 0x20000007 0x0 ] (X>=0 sharedMemory )  [0x7f80f9502ef0] bci=[33,0,39] rc=1 vc=1186 vn=- li=- udi=- nc=0 flg=0x100
n3404n        iconst 0x40200000                                                               [0x7f80f95b27c0] bci=[33,0,39] rc=1 vc=1186 vn=- li=- udi=- nc=0
n3405n      iconst 0                                                                          [0x7f80f95b2810] bci=[33,0,39] rc=1 vc=1186 vn=- li=- udi=- nc=0
```

The problem is that the code in `isFinalizableInlineTest` that performs pattern matching for `isFinalizable` does not expect the `l2i` opcode in the tree in 64-bit mode.  The `l2i` operation is introduced by tree simplification.

I am testing out a fix.

**Comment by hzongaro** [comment]
Fixed in pull request 4891

**Comment by pshipton** [comment]
@hzongaro please create a PR with the fix against the 0.13 branch.

**Comment by hzongaro** [comment]
@pshipton Sorry about that, Peter.  Created pull request #5009 against the 0.13 branch.
